modules/ROOT/pages/secure-configuration-properties.adoc

```diff
@@ -107,58 +76,240 @@ testPropertyA=testValueA
+
+Note that key sizes might depend on the Java Cryptography Extension (JCE) version. Since version 1.8.0_161, the AES key size needs to be of at least 32 characters (256 bytes), instead of 16 characters in previous versions.
+
+[WARNING]
+When you use encrypted properties, it is especially important that you secure access to the operating system. Anyone who can run a `ps` command or view a Java console will be able to see the decrypted values that are stored in the Mule app's memory.
+
+=== Encrypt or Decrypt Single Values
+
+Use the `string` mode to encrypt or decrypt a single value provided as a parameter. Run the following command:
+
+[source,console.linenums]
+----
+java -jar secure-properties-tool.jar string <encrypt|decrypt> <algorithm> <mode> <key> <value>
+----
+
+Replace `<value>` with any desired string value you want to process, for example:
+
+[source,console.linenums]
+----
+java -jar secure-properties-tool.jar string encrypt Blowfish CBC mulesoft "some value to encrypt"
+----
+
+=== Encrypt or Decrypt Files
 
-Here is a flow that uses the encrypted `encrypted.value1` value to set the payload:
+You can use the secure properties tool to encrypt `.yaml` or `.properties` files.
+
+* Use the `file` mode to encrypt all values inside a `.properties` file. The output is a file with the same properties, but its values are encrypted. Run the following command:
++
+[source,console.linenums]
+----
+java -jar secure-properties-tool.jar file <encrypt|decrypt> <algorithm> <mode> <key> <input file> <output file>
+----
++
+For example, assume that this is the input file:
++
+`example_in.yaml`
++
+[source,yaml,linenums]
+----
+properties:
+  example1:
+    value: "Muleman is here"
+  example2: "Max the mule"
+----
++
+If you run the command:
++
+----
+java -jar secure-properties-tool.jar file encrypt Blowfish CBC mulesoft example_in.yaml example_out.yaml
+----
++
+The output file is:
++
+`example_out.yaml`
++
+[source,yaml,linenums]
+----
+properties:
+  example1:
+    value: "![qCReIPK3jcqD7WR84ISSIQ==]"
+  example2: "![En8lII21ZHrdIaINw0+mSA==]"
+----
+
+* Use the `file-level` mode to completely encrypt the properties file. The output is a file that is entirely encrypted.
+
+Using as input the `example_in.yaml` file from the previous example, run the following command:
+----
+java -jar secure-properties-tool.jar file-level encrypt Blowfish CBC mulesoft example_in.yaml example_out.yaml
+----
+
+The output file is: +
+`example_out.yaml`
+
+[source,txt,linenums]
+----
+k/J1EHyxJagOcyQELoaqIDCzzKc1bLGJiHY2dv6jzuhtVpP1LTlgK3y8EH7OkK5iuuu8kik3P5dCDMixZXwn/EV6xbUwLFKnEb4x1pUX6aopLP/KULm0pQ==
+----
+
+== Configure the File with the Mule Secure Configuration Properties Extension Module
```

Comment:
This title could be shortened to just: 
== Configure the File in the Project,  or 
== Configure the File in the Mule Application 

The section explains that this is achieved using the extension Module

Reply:
Updated.